pkg/providers/tinkerbell/upgrade.go

```diff
@@ -238,3 +275,33 @@ func (p *Provider) isScaleUpDown(currentSpec *cluster.Spec, newSpec *cluster.Spe
 
 	return false
 }
+
+func (p *Provider) validateMachineConfigImmutability(ctx context.Context, cluster *types.Cluster, newConfig *v1alpha1.TinkerbellMachineConfig, clusterSpec *cluster.Spec) error {
+	prevMachineConfig, err := p.providerKubectlClient.GetEksaTinkerbellMachineConfig(ctx, newConfig.Name, cluster.KubeconfigFile, clusterSpec.Cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if newConfig.Spec.OSFamily != prevMachineConfig.Spec.OSFamily {
+		return fmt.Errorf("spec.osFamily os immutable. Previous value %v,   New value %v", prevMachineConfig.Spec.OSFamily, newConfig.Spec.OSFamily)
+	}
+
+	if newConfig.Spec.Users[0].SshAuthorizedKeys[0] != prevMachineConfig.Spec.Users[0].SshAuthorizedKeys[0] {
```

Comment:
any reason we are checking only the first key and first user? Dont we want to check them all? 
There should be a function for it 
https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_types.go#L423

Reply:
> EKS-A has only ever supported one User and one sshAuthorizedKey. You will see everywhere in the code that we only deal with the first element [0] of those slices. Looks like there used to be a pre-flight validation to ensure that only one user/key was specified in that slice in the cluster config, but has now disappeared. So now, although multiple users/keys would be accepted by EKS-A, it will silently only use one. That preflight validation needs to come back until multiple users/keys support is implemented and tested.